test/session_pool/session_pool.test.js

```diff
@@ -116,24 +116,24 @@ describe('SessionPool - testing session pool', () => {
             expect(isCalled).toBe(true); //eslint-disable-line
         });
 
-        test('should delete picked session when it is usable a create a new one', async () => {
+        test('should delete picked session when it is unusable and create a new one', async () => {
             sessionPool.maxPoolSize = 1;
             await sessionPool.getSession();
             const session = sessionPool.sessions[0];
 
             session.errorScore += session.maxErrorScore;
             let isCalled = false;
-            const oldRemove = sessionPool._removeSession; //eslint-disable-line
+            const oldRemove = sessionPool._removeRetiredSessions; //eslint-disable-line
 
-            sessionPool._removeSession = (session) => { //eslint-disable-line
+            sessionPool._removeRetiredSessions = (session) => { //eslint-disable-line
                 isCalled = true;
                 return oldRemove.bind(sessionPool)(session);
             };
 
             await sessionPool.getSession();
 
-                expect(isCalled).toBe(true); //eslint-disable-line
-                expect(sessionPool.sessions[0].id === session.id).toBe(false); //eslint-disable-line
+            expect(isCalled).toBe(true);
+            expect(sessionPool.sessions[0].id === session.id).toBe(false);
```

Comment:
please check also that the Map is correctly being emptied to prevent memory leaks

Reply:
added 2 tests to check that map is correctly populated and erased